### PR TITLE
feat(swap-deadline): higher swap deadline

### PR DESCRIPTION
**Pull request by alfetopito**
# Update 2024-10-22

- Split max deadline for EOA and SC wallets:
  - EOA is back at 3h
  - SC set to 12h

## Testing

1. With EOA, edit SWAP deadline
* Should not allow more than 180 min
2. With SC wallets, edit SWAP deadline
* Should allow up to 720 min

# Summary

Part of https://github.com/cowprotocol/cowswap/issues/4321
#208 

Increase SWAP deadline

- Currently set to 12h (up from 30min)
- Currently for all wallets, not sure whether should limit it to only SC wallets
- Moving all quote requests to using the relative `validFor` instead of the absolute `validTo` 

~And since all times will now be relative, we might get rid of this warning:~
This still works, will keep it
image

# To Test

1. On SWAP, pick an expiration > 30 and < 12h (720 min)
* Should place order as usual
4. Alter your system clock and perform the tests from https://github.com/cowprotocol/cowswap/pull/4236
* Should work as described

### CODE REVIEW

File: apps/cowswap-frontend/src/common/updaters/orders/UnfillableOrdersUpdater.ts [hunk 1]
```diff
@@ -222,13 +221,8 @@ async function _getOrderPrice(chainId: ChainId, order: Order, strategy: PriceStr
   }
 
   const legacyFeeQuoteParams = quoteParams as LegacyFeeQuoteParams
-  // Limit order may have arbitrary validTo, but API doesn't allow values greater than 1 hour
-  // To avoid ExcessiveValidTo error we use PRICE_QUOTE_VALID_TO_TIME
-  if (order.class === 'limit') {
-    legacyFeeQuoteParams.validFor = Math.round(PRICE_QUOTE_VALID_TO_TIME / 1000)
-  } else {
-    legacyFeeQuoteParams.validTo = timestamp(order.validTo)
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/common/updaters/orders/UnfillableOrdersUpdater.ts:230]: `validTo` has been removed completely for quote requests. Always use `validFor` for everything now.
Quote responses still have it, as we use that as source of truth for current time.

File: apps/cowswap-frontend/src/modules/tradeWidgetAddons/containers/TransactionSettings/index.tsx [hunk 2]
```diff
@@ -48,7 +48,7 @@ import useNativeCurrency from 'lib/hooks/useNativeCurrency'
 
 import * as styledEl from './styled'
 
-const MAX_DEADLINE_MINUTES = 180 // 3h
+const MAX_DEADLINE_MINUTES = 60 * 12 // 12h
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/tradeWidgetAddons/containers/TransactionSettings/index.tsx:51]: Max SWAP order deadline!!

**Comment by elena-zh** [comment]
Hey @alfetopito , works great!
However, the main issue I've noticed that all market orders have 'partially fillable' type now that is resulting in 99.99% order filling, including [eth-flow orders](https://dev.explorer.cow.fi/sepolia/orders/0x3043f2d0e9254ba080b6f7b5fc74567b1573d288073aaecdeaba399d5ff44f922671994c7d224ac4799ac2cf6ef9ef187d42c69fffffffff?tab=overview)
it is filled
tx completed
image
image

One more limitation that may cause difficulties with signing orders: when connected using WC, I got this error message in 5 minutes. So even if we set a quote validity longer than 5-10 minutes, users still face some limitations from other places. 
5 minutes expired request


----

As for #4236 , it works as expected: orders can be placed, however, on the UI they won't be displayed correctly. Actually, that is why we show the warning at the top: 
- limit order that is placed rn, but shows creation date 4 hours ago
ago
- placed swap order with duration < than a time difference: 
expired on placement
 - open order is labeled ad expired:
filled

**Comment by alfetopito** [comment]
> Hey @alfetopito , works great! However, the main issue I've noticed that all market orders have 'partially fillable' type now that is resulting in 99.99% order filling, including [eth-flow orders](https://dev.explorer.cow.fi/sepolia/orders/0x3043f2d0e9254ba080b6f7b5fc74567b1573d288073aaecdeaba399d5ff44f922671994c7d224ac4799ac2cf6ef9ef187d42c69fffffffff?tab=overview) it is filled tx completed image image
> 

As mentioned on slack, unrelated issue, fixed on https://github.com/cowprotocol/cowswap/pull/5016

> One more limitation that may cause difficulties with signing orders: when connected using WC, I got this error message in 5 minutes. So even if we set a quote validity longer than 5-10 minutes, users still face some limitations from other places. 5 minutes expired request

That's unrelated to this change. Quote validity changes were purely backend. This (WC disconnection) is a different issue, which we do have a GH issue for already.

> As for #4236 , it works as expected: orders can be placed, however, on the UI they won't be displayed correctly. Actually, that is why we show the warning at the top:

Then, both parts are working as expected \o/

>     * limit order that is placed rn, but shows creation date 4 hours ago
>       ago
> 
>     * placed swap order with duration < than a time difference:
>       expired on placement
> 
>     * open order is labeled ad expired:
>       filled

**Comment by alfetopito** [comment]
I'll make sure the extended time only works for SC wallets and let you know

**Review by elena-zh — APPROVED** [review]
Great, @alfetopito , than k you!

Btw, should we mentioned somewhere in the docs about this change?

**Comment by alfetopito** [comment]
> Great, @alfetopito , than k you!
> 
> Btw, should we mentioned somewhere in the docs about this change?

I don't know. Do we have anything about that in the docs already?

Doesn't seem like it https://docs.cow.fi/search?q=deadline

In that case, I wouldn't mention it.
It would make more sense IMO to show that in the UI via tooltips/error messages, but it's out of the scope of the current change.

**Review by shoom3301 — APPROVED** [review]

**Pull request merged**